src/st/dai/e_bone_pillar.c

```diff
@@ -0,0 +1,489 @@
+                CreateEntityFromCurrentEntity(E_BONE_PILLAR_HEAD, entity);
+                entity->params = 1;
+                entity->posX.i.hi = self->posX.i.hi;
+                entity->posY.i.hi = self->posY.i.hi;
+            }
+            SetStep(2);
+        }
+        break;
+    case 0x2:
+        if (GetDistanceToPlayerX() < 0x70) {
+            SetStep(3);
+        }
+        break;
+    case 0x3:
+        if (!self->step_s) {
+            self->ext.et_bonePillar.unk80 =
+                D_us_80181D30[self->ext.et_bonePillar.unk84];
+            self->ext.et_bonePillar.unk84++;
+            self->ext.et_bonePillar.unk84 &= 7;
+            self->step_s++;
+        }
+        if (self->params) {
+            self->animCurFrame = 9;
+        } else {
+            self->animCurFrame = 1;
+        }
+        if (self->params == ((GetSideToPlayer() & 1) ^ 1)) {
+            if (!--self->ext.et_bonePillar.unk80) {
+                SetStep(4);
+            }
+        }
+        if (GetDistanceToPlayerX() > 0x80) {
+            SetStep(3);
+        }
+        break;
+    case 0x4:
+        if (g_Timer & 2) {
+            self->palette = g_EInitBonePillarHead[3];
+        } else {
+            self->palette = g_EInitBonePillarHead[3] + 1;
+        }
+        if (self->params) {
+            unkVar = AnimateEntity(&D_us_80181CE4, self);
+        } else {
+            unkVar = AnimateEntity(&D_us_80181CAC, self);
+        }
+        if (!unkVar) {
+            self->palette = g_EInitBonePillarHead[3] + 1;
+            SetStep(3);
+        }
+        if ((self->pose == 0x11) && (unkVar & 0x80)) {
+            entity = AllocEntity(&g_Entities[160], &g_Entities[192]);
+            if (entity != NULL) {
+                CreateEntityFromEntity(E_BONE_PILLAR_FIRE, self, entity);
+                entity->facingLeft = self->params;
+                if (self->params) {
+                    entity->posY.i.hi -= 0xE;
+                    break;
+                }
+                entity->posY.i.hi += 8;
+            }
+        }
+        break;
+    case 0x8:
+        ptr = D_us_80181C8C;
+        unkVar = 0;
+        if (!self->params) {
+            ptr += 4;
+            unkVar = 2;
+        }
+        for (i = 0; i < 2; i++) {
+            entity = AllocEntity(&g_Entities[224], &g_Entities[256]);
+            if (entity != NULL) {
+                CreateEntityFromEntity(E_BONE_PILLAR_PIECES, self, entity);
+                entity->params = i + unkVar;
+                entity->posX.i.hi += *ptr++;
+                entity->posY.i.hi += *ptr++;
+            }
+        }
+        entity = AllocEntity(&g_Entities[224], &g_Entities[256]);
+        if (entity != NULL) {
+            CreateEntityFromEntity(E_EXPLOSION, self, entity);
+            entity->params = 2;
+        }
+        if ((!self->params) &&
+            ((entity = self + 1, (entity->entityId) == E_BONE_PILLAR_HEAD) ||
+             ((entity->entityId) == E_BONE_PILLAR_SPIKE_BALL))) {
```

Comment:
Do you have a scratch for this? The assign and condition within an `if`, while valid is pretty uncommon an likely to be a more simple assignment enclosed within the `if`. Should be easier to read that way also.

eg. something like

```c
if (!self.params) {
   entity = self + 1;
   if (entity->entityId == E_BONE_PILLAR_HEAD || entity->entityId == E_BONE_PILLAR_SPIKE_BALL) {
     entity->ext.et_bonePillar.unk85 = 1;
  }
}
```

Reply:
This one was a little difficult for me.  I felt like it was overly complex, but I couldn't figure out a better way.  Since it is written this way in e_fishhead, that's what I went with.  Your suggestion worked fine, though, and I'll be pushing it up with the other review adjustments.

I also updated the description of this PR with the scratches.